TROUBLESHOOTING.md

```diff
@@ -33,20 +36,103 @@ However, if you need to declare dependency on gRPC yourself, we recommend doing
   </dependencies>
 </dependencyManagement>
 ```
-so that versions cannot go out of sync.
 
-See [gRPC documentation](https://github.com/grpc/grpc-java/blob/master/SECURITY.md#troubleshooting)
-for more details.
+The `grpc-netty-shaded` dependency avoids conflicts with other `netty` dependencies that may also be in the classpath.
+
+If you are using `google-cloud-java` version 0.35.0 or above, then it already uses `grpc-netty-shaded`. If you are still running into `ALPN` related problems, please see [gRPC Troubleshooting guide](https://github.com/grpc/grpc-java/blob/master/SECURITY.md#troubleshooting) for other causes.
+
+## ClassNotFoundException, NoSuchMethodError, NoClassDefFoundError
+
+These errors are usually caused by having multiple versions or conflicting versions of the same dependency in the classpath.
+Usually these dependency conflicts occur with `guava` or `protobuf-java`.
+
+There may be multiple sources for classpath conflicts:
+- Multiple transitive dependencies in the dependency tree
+- Runtime classpath has different versions of dependences than what you specified in the build
+
+For example, if you have a direct dependency or a transitive dependency on Guava version 10.0, and also `google-cloud-java` uses Guava version 20.0, then `google-cloud-java` could be using Guava methods that don't exist in Guava 10.0, and causing `NoSuchMethodError`.
+
+Similarily, if your classpath has older version of `protobuf-java`, but `google-cloud-java` requires newer version, then you may see `NoClassDefFoundError` that fails to initialize `google-cloud-java` classes, e.g.:
+
+```
+java.lang.NoClassDefFoundError: Could not initialize class com.google.pubsub.v1.PubsubMessage$AttributesDefaultEntryHolder
+```
+
+### Validate the conflict
+
+Check the dependency tree to see if you have multiple versions of the same dependencies, e.g.:
+
+```
+$ mvn dependency:tree
+```
+
+Look for versions of all of the `guava`, `protobuf-java`, etc.
```

Comment:
s/Look for versions of all of the/Look for versions of potentially conflicting dependencies like/

Reply:
done